### PR TITLE
Use nix check for CI testing

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

File: bin/assigner/CMakeLists.txt [hunk 1]
```diff
@@ -41,35 +41,18 @@ target_link_libraries(${CURRENT_PROJECT_NAME}
...
                       ${Boost_LIBRARIES})
 
 target_include_directories(${CURRENT_PROJECT_NAME} PUBLIC
                            $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
                            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-                           ${CMAKE_CURRENT_SOURCE_DIR}/../../libs/circifier/llvm/include
-                           ${CMAKE_CURRENT_SOURCE_DIR}/../../libs/circifier/utils/bazel/llvm-project-overlay/llvm/include
+                           ${CMAKE_SOURCE_DIR}/libs/circifier/llvm/include
+                           ${CMAKE_SOURCE_DIR}/libs/circifier/utils/bazel/llvm-project-overlay/llvm/include
+                           ${CMAKE_SOURCE_DIR}/libs/assigner/include
```
Comment (akokoshn) [inline bin/assigner/CMakeLists.txt:53]: use `crypto3::assigner`

File: bin/test_package.cpp
Comment (aleasims) [inline bin/test_package.cpp:1]: What is this?

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ nix-3rdparty.overlays.${system}.default ];
+        };
+        crypto3 = nil-crypto3.packages.${system}.default;
+        blueprint = nil-zkllvm-blueprint.packages.${system}.default;
+
+      in {
+        packages = rec {
+          inherit pkgs crypto3 blueprint;
```
Comment (aleasims) [inline flake.nix:51]: This is a copy-paste mistake from our previous discussion.

File: cmake/CheckSSE.cmake [hunk 3]
```diff
@@ -59,7 +59,7 @@ macro(check_sse)
         endif(SSE42_TRUE)
 
     elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
-        exec_program("/usr/sbin/sysctl -n machdep.cpu.features" OUTPUT_VARIABLE
+        exec_program("/var/empty/sbin/sysctl -n machdep.cpu.features" OUTPUT_VARIABLE
```
Comment (makxenov) [inline cmake/CheckSSE.cmake:62]: Is this file used anywhere?

File: cmake/create-symlinks.cmake [hunk 4]
```diff
@@ -5,7 +5,7 @@ list(APPEND LLVM_BINARIES clang llvm-link)
 
 # Link directory would differ for cases when we have DESTDIR from CPack or simple installation
 if (DEFINED ENV{DESTDIR})
-    set(LINK_DESTDIR $ENV{DESTDIR}/usr/bin)
+    set(LINK_DESTDIR $ENV{DESTDIR}/var/empty/bin)
```
Comment (makxenov) [inline cmake/create-symlinks.cmake:8]: AFAIR the point of these symlinks was having zkllvm binaries in PATH. In particular, CircuitCompile.cmake [relies](https://github.com/NilFoundation/zkLLVM/blob/eb864384bf902c6a2f3a60965bc2a3d6f3338a8a/cmake/CircuitCompile.cmake#L71) on this presence. If we can't make Nix work with `/usr/bin` directory, then we have to rewrite CircuitCompile file, and the symlinks will become unnecessary. Or I would suggest just set `dontInstall = true;` for now.

File: tests/cpp/CMakeLists.txt [hunk 5]
```diff
@@ -17,13 +17,8 @@ add_custom_target(all_tests_run_expected_res_calculation)
 add_custom_target(all_tests_assign_circuits)
 add_custom_target(all_tests_run)
 
-if(NOT Boost_UNIT_TEST_FRAMEWORK_FOUND)
-    cm_find_package(Boost REQUIRED COMPONENTS unit_test_framework filesystem)
-endif()
-
 cm_test_link_libraries(
-                       crypto3::zk
-                       crypto3::assigner
```
Comment (makxenov) [inline tests/cpp/CMakeLists.txt:26]: Please restore `crypto3::assigner` and remove additional include dir (same as for `bin/assigner/CMakeLists.txt`)

File: tests/move_assigner_outputs.sh [hunk 6]
```diff
@@ -0,0 +1,11 @@
+SRC_DIR="$1"
```
Comment (makxenov) [inline tests/move_assigner_outputs.sh:1]: This file seems unused

File: tests/copy_assigner_outputs.sh [hunk 7]
```diff
@@ -1,5 +1,5 @@
-SRC_DIR="build/examples/cpp"
-DEST_DIR="$1"
+SRC_DIR="$1"
+DEST_DIR="$2"
```
Comment (makxenov) [inline tests/copy_assigner_outputs.sh:2]: Can't find actual change of usage with two arguments instead of one.

File: flake.nix (hunk 2 above)
Comment (AndreyMlashkin) [reply flake.nix:51]: corrected as in the previous PR

File: bin/test_package.cpp
Comment (AndreyMlashkin) [reply bin/test_package.cpp:1]: That shall be reverted, a8b0939c451ec1510969a8aea3a90d24da8310e1
"add test exectutable to speedup testing. TODO: revert"

File: cmake/create-symlinks.cmake (hunk 4 above)
Comment (AndreyMlashkin) [reply cmake/create-symlinks.cmake]: I have re-written it as set(LINK_DESTDIR $ENV{DESTDIR}/usr/bin)

File: tests/cpp/CMakeLists.txt (hunk 5 above)
Comment (AndreyMlashkin) [reply tests/cpp/CMakeLists.txt:26]: Done

File: tests/move_assigner_outputs.sh (hunk 6 above)
Comment (AndreyMlashkin) [reply tests/move_assigner_outputs.sh]: Removed

File: tests/copy_assigner_outputs.sh (hunk 7 above)
Comment (AndreyMlashkin) [reply tests/copy_assigner_outputs.sh]: Reverted

**Review by akokoshn — APPROVED** [review]

**Pull request merged**